### DISCUSSION TITLE
Public server availability on AWS EC2

### DISCUSSION THREAD

**Discussion by PatchworkPirate** [cpvalente/ontime#728] [Q&A] [answered]
Hi,

I'm trying to run the application on an AWS EC2 instance so that the cuesheets can be available to those who are working on the shows remotely. When I start up the application it registers on 127.0.0.1 and 172.31.14.65 but I can't seem to access the server via it's publicly available IP on the default port of 4001. Strangely I can access the web control panel of vMix so the unit is definitely accessible and open to traffic. When I try to access Ontime remotely the browser just loads endlessly with no response.

Is there a setting that I need to change to allow it to be openly available?

**Comment by cpvalente** [comment]
Hey, thank you for reaching out.
I will try to provide some help here, but it would be tricky for me to troubleshoot your setup.
There are a few users with similar setups so I know the setup is possible

Few questions:
Which app version are you using?
Are you using the docker container? (you probably should)

**Comment by PatchworkPirate** [comment]
No worries, I'll try to provide as much detail as possible.

I'm running v1.2.1 on an AWS g4dn.8xLarge EC2 unit, it uses a GPU so I can use the unit like a normal desktop and access the GUI. I'm not using docker but can definitely switch to using it if it would make things more consistent and easier to troubleshoot for sure. Let me know what other information you need.

On a slightly different note, I use cloud based machines and services all the time for live broadcast so in response to your Reddit thread I'd be more than happy to help test out builds in these workflows when you need.

**Comment by je71175** [comment, marked as answer]
Hi - here is my list of instructions for getting onTime working on AWS.  I also installed additional apps other than ontime, so you can disregard those AWS network rules 

Setup work
	1. Install SSH - Apps, optional features
	2. Enable SSH Services (Task Manager)

FTP files across to AWS
	1. Install apps
	2. Allow through Defender firewall


AWS Network Inbound rules

Port range	Protocol	Source	Security groups
22	TCP	0.0.0.0/0	launch-wizard-1
3389	TCP	0.0.0.0/0	launch-wizard-1
4001	TCP	0.0.0.0/0	launch-wizard-1 Ontime
9993	USP	0.0.0.0/0	Zero Tier
22000	TCP	0.0.0.0/0	launch-wizard-1 SYNC Thing
22000	UDP	0.0.0.0/0	launch-wizard-1  SYNC Thing

**Comment by PatchworkPirate** [comment]
> Hi - here is my list of instructions for getting onTime working on AWS. I also installed additional apps other than ontime, so you can disregard those AWS network rules
> 
> Setup work 1. Install SSH - Apps, optional features 2. Enable SSH Services (Task Manager)
> 
> FTP files across to AWS 1. Install apps 2. Allow through Defender firewall
> 
> AWS Network Inbound rules
> 
> Port range Protocol Source Security groups 22 TCP 0.0.0.0/0 launch-wizard-1 3389 TCP 0.0.0.0/0 launch-wizard-1 4001 TCP 0.0.0.0/0 launch-wizard-1 Ontime 9993 USP 0.0.0.0/0 Zero Tier 22000 TCP 0.0.0.0/0 launch-wizard-1 SYNC Thing 22000 UDP 0.0.0.0/0 launch-wizard-1 SYNC Thing

Ah amazing, was just Windows Defender blocking it by default. Thanks for sending those instructions over!

**Comment by cpvalente** [comment]
Thank you @je71175  for stepping in.
Either way @PatchworkPirate I would still recommend looking at the docker container. It a much lighter footprint in your AWS infrastructure
See the project and the docker compose file in docker hub 
https://hub.docker.com/r/getontime/ontime
